### ISSUE TITLE
improve pretty print escaping

### ISSUE THREAD

**Issue by bradrydzewski** [drone/drone-yaml#45]
conversion of the following 0.8 yaml:

```
  job:
    image: my/image
    param: 'One: Two'
```

leads to an invalid yaml due to lack of escaping:

```
- name: job
  pull: default
  image: my/image
  settings:
    param: One: Two
```

my guess is that we need to add `:` to the following function to ensure values get quoted:
https://github.com/drone/drone-yaml/blob/master/yaml/pretty/util.go#L66

**Comment by bradrydzewski** [comment]
fixed cc @jimsheldon

**Comment by bradrydzewski** [comment]
re-opening because this causes all values with `:` to be quoted and thus caused unit tests to fail. 

```
image: "alpine:3.8"
```

We should update to only quote a `:` if whitespace precedes or follows the character. For example we should apply a regular expression like `\s:`

**Comment by bradrydzewski** [comment]
fixed by 3ea0d8d

**Comment by jimsheldon** [comment]
Verified the above fix 👍
